src/github.com/getlantern/flashlight/analytics/analytics_test.go

```diff
@@ -0,0 +1,30 @@
+package analytics
+
+import (
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/getlantern/eventual"
+	"github.com/getlantern/golog"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestAnalytics(t *testing.T) {
+	logger := golog.LoggerFor("flashlight.analytics_test")
+
+	params := eventual.NewValue()
+	start("1", "2.2.0", func(time.Duration) string {
+		return "127.0.0.1"
+	}, 1*time.Millisecond, func(args string, addr eventual.Getter) {
+		logger.Debugf("Got args %v", args)
+		params.Set(args)
+	})
+
+	args, ok := params.Get(40 * time.Second)
+	assert.True(t, ok)
+
+	argString := args.(string)
```

Comment:
This will fail if we don't get a value in `params` on time, but being a test we have control over that time and it is perfectly fine.